pkg/proxy/ipvs/proxier.go

```diff
@@ -720,51 +720,58 @@ func (handle *LinuxKernelHandler) GetKernelVersion() (string, error) {
 }
 
 // CanUseIPVSProxier checks if we can use the ipvs Proxier.
-// This is determined by checking if all the required kernel modules can be loaded. It may
-// return an error if it fails to get the kernel modules information without error, in which
-// case it will also return false.
-func CanUseIPVSProxier(handle KernelHandler, ipsetver IPSetVersioner, scheduler string) error {
-	mods, err := handle.GetModules()
+// If the ipvs already has virtual servers (VS) we assume this is a re-start and
+// skip the tests. If the ipvs is empty we try to add a VS with the passed scheduler.
+// If that works we assume that ipvs and the passed scheduler are supported.
+// The ipset version is also checked.
+func CanUseIPVSProxier(ipvs utilipvs.Interface, ipsetver IPSetVersioner, scheduler string) error {
+	// Check is any virtual servers (vs) are configured. If any, we assume
+	// that this is a kube-proxy re-start and skip the checks.
+	vservers, err := ipvs.GetVirtualServers()
 	if err != nil {
-		return fmt.Errorf("error getting installed ipvs required kernel modules: %v", err)
+		klog.ErrorS(err, "Can't read the ipvs")
+		return err
 	}
-	loadModules := sets.NewString()
-	loadModules.Insert(mods...)
-
-	kernelVersionStr, err := handle.GetKernelVersion()
-	if err != nil {
-		return fmt.Errorf("error determining kernel version to find required kernel modules for ipvs support: %v", err)
+	klog.V(5).InfoS("Virtual Servers", "count", len(vservers))
+	if len(vservers) > 0 {
+		klog.InfoS("Assuming kube-proxy re-start", "virtual-servers", len(vservers))
+		return nil
 	}
-	kernelVersion, err := version.ParseGeneric(kernelVersionStr)
-	if err != nil {
-		return fmt.Errorf("error parsing kernel version %q: %v", kernelVersionStr, err)
+
+	// BUG: If ipvs is not compiled into the kernel ipvs.GetVirtualServers
+	// does not return an error. Instead also ipvs.AddVirtualServer returns OK
+	// (no error)!
+
+	// This is the first time kube-proxy is loaded on the node. Try to
+	// insert a dummy VS with the passed scheduler. The VIP address can be
+	// anything since no traffic will be routed to ipvs yet.
+	vs := utilipvs.VirtualServer{
```

Comment:
I don't know about this - there's not really any magic IP or port that we can use which is guaranteed NOT to collide with existing services.  Is there a different protocol we can use or some other way to be SURE that it isn't really used?  And what if it exists but was a leftover from a previous crash?  Maybe you should force-delete first, ignoring errors, then create (assuming we keep this mechanism).

Reply:
The ipvs is empty when this is executed (no virtual servers). To get traffic to it the address must be assigned to interface `kube-ipvs0`, so it _should_ be safe.

But I can surely find a more obscure test address. For instance an ipv6 address reserver in [fc3849](https://www.rfc-editor.org/rfc/rfc3849) like 2001:DB8::3333 which should *never* be used.

Reply:
That would help (though beware nodes without IPv6?) - a great big comment here about why this is "safe" would be nice.